src/index.ts

```diff
@@ -156,7 +156,7 @@ export function useContextSelector<Value, Selected>(
   const contextValue = useContextOrig(
     context as unknown as ContextOrig<ContextValue<Value>>,
   )[CONTEXT_VALUE];
-  if (typeof process === 'object' && process.env.NODE_ENV !== 'production') {
+  if (process.env.NODE_ENV !== 'production') {
```

Comment:
There's a reason that exists there. In some cases, `process` can be undefined, and `process.env.NODE_ENV` throws an error, because `process` is undefined...

Hmm, I don't think there's a solution to fill all requirements.